src/sync/app_credentials.hpp

```diff
@@ -21,6 +21,7 @@
 
 #include <functional>
 #include <string>
+#include <object-store/src/util/bson/bson.hpp>
```

Comment:
This include should not be here and we should be forward declaring BsonDocument.

```cpp
namespace realm {
namespace bson {
class Bson;
template <typename> class IndexedMap;
using BsonDocument = IndexedMap<Bson>;
}
namespace app {
...
```

And then the method signature should be:
```cpp
static AppCredentials function(bson::BsonDocument& payload_json)
```